ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java

```diff
@@ -77,9 +84,7 @@ private GPGNetClient(Socket socket) {
             } catch (IOException e) {
                 log.error("Could not create GPGNet output steam to FA", e);
             }
-
-            listenerThread = new Thread(this::listenerThread);
-            listenerThread.start();
+            listener = CompletableFuture.runAsync(this::listenerThread, IceAdapter.getExecutor());
```

Comment:
Same here. Why not just spawn a virtual thread directly?

Reply:
You have a reference to the thread so you can set the name, exceptionHandler and interrupt it.